File: QadeerApp.Web/Modules/ServerTypes/Administration/TrainingCourseColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { TrainingCourseRow } from "./TrainingCourseRow";

export interface TrainingCourseColumns {
    TrainingCourseId: Column<TrainingCourseRow>;
    Name: Column<TrainingCourseRow>;
    Code: Column<TrainingCourseRow>;
    DepartmentName: Column<TrainingCourseRow>;
    SpecializationName: Column<TrainingCourseRow>;
    IsActive: Column<TrainingCourseRow>;
}

export class TrainingCourseColumns extends ColumnsBase<TrainingCourseRow> {
    static readonly columnsKey = 'Administration.TrainingCourse';
    static readonly Fields = fieldsProxy<TrainingCourseColumns>();
}

File: QadeerApp.Web/Modules/Administration/TrainingCourse/TrainingCourseColumns.cs
namespace QadeerApp.Administration.Columns;

[ColumnsScript("Administration.TrainingCourse")]
[BasedOnRow(typeof(TrainingCourseRow), CheckNames = true)]
public class TrainingCourseColumns
{
    [EditLink, AlignRight, Width(60)]
    public int TrainingCourseId { get; set; }

    [EditLink, Width(220)]
    public string Name { get; set; }

    [Width(120)]
    public string Code { get; set; }

    [Width(180), QuickFilter]
    public string DepartmentName { get; set; }

    [Width(180), QuickFilter]
    public string SpecializationName { get; set; }

    [Width(80), QuickFilter]
    public short IsActive { get; set; }
}
